subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/StartParameterResolutionOverride.java

```diff
@@ -104,7 +104,7 @@ public DependencyVerificationOverride dependencyVerificationOverride(BuildOperat
                                                                          BuildCommencedTimeProvider timeProvider,
                                                                          Factory<GradleProperties> gradlePropertiesFactory) {
         List<String> checksums = startParameter.getWriteDependencyVerifications();
-        if (!checksums.isEmpty()) {
+        if (!checksums.isEmpty() || startParameter.isExportKeys()) {
```

Comment:
It makes sense to me that a user that passes --export-keys without --write-verification-metadata expects the keys to be exported

I'm less confident that in that case we also want to skip validating the verification metadata too ( https://github.com/gradle/gradle/issues/18567#issuecomment-1175643973 ), which is a side-effect of this change. It does make sense to me that a user passing --export-keys is still in the process of updating the state of what is trusted so it may be reasonable to also skip verifying the verification metadata in that case.

Hopefully this change at least makes the behavior less surprising?

Reply:
I think it's ok we skip metadata verification. I imagine it's expected, since this method is similar to `--write-verification-metadata`. Also currently when you currently export keys metadata is not verified, so it seems to me we would have a different if you pass `--write-verification-metadata` and `--export-keys` or just `--export-keys` which is not good.

But it's a good point, I think we should at least document that `--export-keys` behaves the same way as `--write-verification-metadata`.

Reply:
Thanks!

I think I've now updated subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/dependency_verification.adoc